### PR TITLE
chore(types): Add type hints for mock to dev deps

### PR DESCRIPTION
**Pull request by wedamija**

### CODE REVIEW

File: requirements-dev.txt [hunk 1]
```diff
@@ -33,6 +33,7 @@ mypy>=0.982
 types-beautifulsoup4
 types-freezegun
 types-jsonschema
+types-mock
```
Comment (asottile-sentry) [inline requirements-dev.txt:36]: we should not be using `mock` -- we should be using `unittest.mock`

**Review by asottile-sentry — CHANGES_REQUESTED** [review]

File: requirements-dev.txt (hunk 1 above)
Comment (asottile-sentry) [reply requirements-dev.txt:36]: I've been sloooowly trying to kill it, some crappy third party dependency pulls it in still though 😭

Comment (asottile-sentry) [reply requirements-dev.txt:36]:
```
$ grep -i '^requires-dist.*mock[^;]*$' .venv/lib/python3.8/site-packages/*/METADATA
.venv/lib/python3.8/site-packages/exam-0.10.6.dist-info/METADATA:Requires-Dist: mock
```

bleh

Comment (wedamija) [reply requirements-dev.txt:36]: Lol, `exam` was built at Disqus and I think the guy who wrote/maintains it now mostly works in rubby. So we'll have to fork it ourselves if we want to fix it.

Comment (asottile-sentry) [reply requirements-dev.txt:36]: yeahhhh I've just been slowly factoring it out instead

**Pull request closed without merging**